pandas/core/categorical.py

```diff
@@ -127,6 +127,8 @@ def maybe_to_categorical(array):
     """ coerce to a categorical if a series is given """
     if isinstance(array, (ABCSeries, ABCCategoricalIndex)):
         return array._values
+    elif isinstance(array, np.ndarray):
```

Comment:
in `NonConsolidatingBlock._unstack()` calling `.make_block_same_class()`, making a `CategoricalBlock` from unstacked `values` (an `ndarray`). Otherwise fails `frame.test_reshape.TestDataFrameReshape.test_unstack_preserve_dtypes`.
Since the function is named `maybe_to_categorical` and accepts argument `array`, the change seems like making perfect sense.